src/enzo/New_Grid_ReadGrid.C

```diff
@@ -223,10 +224,18 @@ int grid::Group_ReadGrid(FILE *fptr, int GridID, HDF5_hid_t file_id,
  
     /* 4) Read gravity info */
  
-    if (SelfGravity)
+    if (SelfGravity) {
       if (fscanf(fptr, "GravityBoundaryType = %"ISYM"\n",&GravityBoundaryType) != 1) {
-		ENZO_FAIL("Error reading GravityBoundaryType.");
+	// If turning on self gravity in a restart, then determine
```

Comment:
This is OK, but after double-checking the code, one slightly cleaner way to do this would be to remove the check here and just set the GravityBoundaryType in Group_ReadDataHierarchy.  You do this anyway for the topgrid, but you could also include subgrids in that by checking to see if ParentGrid != NULL and then set GravityBoundaryType = SubGridIsolated (otherwise set to MetaData.GravityBoundary as you do now). That would remove this slightly awkward calculation here and push the check into the right level. In that case, the error check for reading GravityBoundaryType could just silently fail. This would also have the advantage that the code would work if NEW_GRID_IO were turned off (right now I think it would not since you don't have this bit of code in the old grid reader, Grid_Group_ReadGrid.C).

Reply:
I like this approach a lot better. I've made your suggested change.